src/SFA.DAS.LearnerDataMismatches.Domain/LearnerReport.cs

```diff
@@ -5,15 +5,12 @@ namespace SFA.DAS.LearnerDataMismatches.Domain
 {
     public class LearnerReport
     {
-        private List<Payments.Model.Core.Entities.ApprenticeshipModel> apps2;
-        private List<Payments.Model.Core.Audit.EarningEventModel> earnings;
-
-        public LearnerReport(List<Payments.Model.Core.Entities.ApprenticeshipModel> apprenticeships, List<Payments.Model.Core.Audit.EarningEventModel> earnings)
+        public LearnerReport(
+            List<Payments.Model.Core.Entities.ApprenticeshipModel> apprenticeships,
+            List<Payments.Model.Core.Audit.EarningEventModel> earnings,
+            List<Payments.Model.Core.Audit.DataLockEventModel> locks)
         {
-            this.apps2 = apprenticeships;
-            this.earnings = earnings;
-
-            var apps = apps2
+            var apps = apprenticeships
```

Comment:
It may be that a previous training provider keeps submitting an ILR for an apprentice, even when they have moved to a new provider.  I'm not entirely sure how much we are going to end up showing.

On a more concrete note, I'd like to keep the filtering logic inside the `LearnerReport` domain class as much as possible.  Whilst this may result in a little more data being retrieved than necessary, it does mean that we aren't partitioning the logic into multiple places.

Reply:
As discussed, the filter for `Active` status is already happening in the page, so I have consolidated the filter to one place.